### DISCUSSION TITLE
Trouble with Locale Switching in Next.js 14 using next-intl

### DISCUSSION THREAD

**Discussion by ozgurcoban** [amannn/next-intl#722] [Q&A]
I'm currently working on a project where I'm using next-intl for localization with Next.js 14 and the App Router. This is my first time implementing it, and I've hit a snag.

My default language is set to English ('en'). However, when I'm on the Swedish version of the site ([www.localhost/sv](http://www.localhost/sv)) and try to navigate to the English version by removing 'sv' from the URL, it automatically redirects me back to '/sv'. The only way I can switch to the English version is by explicitly typing '/en' in the URL; otherwise, I just keep bouncing back to the Swedish version.

I've followed the instructions on the official repository and the next-intl website, but I'm still struggling to make it work as expected. Interestingly, the [demo](https://next-intl-example-app-router.vercel.app/en) works as I desire; for example, when on the German version and removing '/de' from the URL, it takes me to the English version.

Changing `localeDetection` to `false` helps, but that removes the ability to detect and set the locale based on the user's preferences.

Is there a good way to work around this problem?

**Comment by amannn** [comment]
Hey, thanks for trying out `next-intl`! This is working as intended, see [the docs on locale detection](https://next-intl-docs.vercel.app/docs/routing/middleware#locale-detection) (cookies take precedence for unprefixed pathnames). To change the locale, navigating to a prefixed route of another locale is the way to go.

Hope this helps!

**Comment by ozgurcoban** [comment]
Thanks, it helped to understand so I can move on :)
